File: c_c++/excelSheetColumnNumber.c
#include <stdio.h>

int titleToNumber(char *s) {
    unsigned ans = 0;
    char * p = s;
    while( p != NULL && *p != '\0' ) {
        if( (*p < 'A') || (*p > 'Z') ) {
            return 0;
        }
        ans = ans * 26 + ( *p - 'A' ) + 1;
        p ++;
    }
    return ans;
}

int main () {

	printf( "%d\n", titleToNumber ( "AA" ) );
	return 0;
}

File: java/src/leetcode_in_java/ExcelSheetColumnNumber.java
package leetcode_in_java;

/**
 * Created by bopang on 15/8/6.
 */
class ExcelSheetColumnNumber {
    public int titleToNumber(String s) {
        if( null == s || 0 == s.length() )
            return 0;
        int ans = 0;
        int base = 1;
        char[] arrStr = s.toCharArray();
        for( int idx = arrStr.length - 1; idx >= 0; idx -- ) {
            ans += ( arrStr[ idx ] - 'A' + 1 ) * base;
            base *= 26;
        }
        return ans;
    }
}
